core/base/src/TApplication.cxx

```diff
@@ -405,7 +405,16 @@ void TApplication::GetOptions(Int_t *argc, char **argv)
          // the web mode is requested
          const char *opt = argv[i] + 5;
          argv[i] = null;
-         gROOT->SetWebDisplay((*opt == '=') ? opt + 1 : "");
+         if (strcmp(opt, "=off") == 0)
+            gROOT->SetWebDisplay("off");
+         else {
+            printf("\nWARNING!\n");
+            printf("Web mode switch --web is disabled for security reason.\n");
+            printf("See https://root.cern/about/security/ for more information.\n");
+            printf("If one really wants to enable web display, one can call\n");
+            printf("gROOT->SetWebDisplay(); in ROOT prompt or in startup scripts\n\n");
```

Comment:
```suggestion
            printf("For environments controlling the security issues you can enable web display by calling\n");
            printf("gROOT->SetWebDisplay(); in ROOT prompt or in startup scripts\n\n");
```
This is the backtrace calling this function for the first time:
```
#0  TApplication::GetOptions at core/base/src/TApplication.cxx:346
#1  0x00007ffff79a5ecc in TApplication::TApplication
    at core/base/src/TApplication.cxx:177
#2  0x00007ffff7f9c6db in TRint::TRint at core/rint/src/TRint.cxx:174
#3  0x00005555555555c6 in main at main/src/rmain.cxx:81
```
Doing anything at the prompt might be too late. I doubt that startup scripts get run early enough?

Reply:
> Why? Default is off and it is ok. And in ROOT prompt user can switch web mode back - if he want.

I don't understand how can that warning be triggered. Can you try in your PR build and show me the sequence of calls you do and where the output happens?

Reply:
> I don't understand how that warning can be triggered.

Calling `root --web`.